File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/custom/LenientTest.java
package com.jaynewstrom.json.sample.custom;

import com.jaynewstrom.json.sample.JsonTestHelper;

import org.junit.Test;

import static org.fest.assertions.api.Assertions.assertThat;

public final class LenientTest {
    @Test public void testLenientSerializerTrue() {
        JsonTestHelper testHelper = new JsonTestHelper();
        String serialized = testHelper.serialize(new Lenient(true));
        assertThat(serialized).isEqualTo("{\"foo\":\"true\"}");
    }

    @Test public void testLenientSerializerFalse() {
        JsonTestHelper testHelper = new JsonTestHelper();
        String serialized = testHelper.serialize(new Lenient(false));
        assertThat(serialized).isEqualTo("{\"foo\":\"false\"}");
    }

    @Test public void testLenientDeserializerTrue() {
        JsonTestHelper testHelper = new JsonTestHelper();
        Lenient lenient = testHelper.deserializeString(Lenient.class, "{\"foo\":\"true\"}");
        assertThat(lenient.foo).isEqualTo(true);
    }

    @Test public void testLenientDeserializerFalse() {
        JsonTestHelper testHelper = new JsonTestHelper();
        Lenient lenient = testHelper.deserializeString(Lenient.class, "{\"foo\":\"false\"}");
        assertThat(lenient.foo).isEqualTo(false);
    }
}

File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/custom/LenientTest.kt
package com.jaynewstrom.json.sample.custom

import com.jaynewstrom.json.sample.JsonTestHelper
import org.fest.assertions.api.Assertions.assertThat
import org.junit.Test

class LenientTest {
    @Test fun testLenientSerializerTrue() {
        val testHelper = JsonTestHelper()
        val serialized = testHelper.serialize(Lenient(true), Lenient::class.java)
        assertThat(serialized).isEqualTo("{\"foo\":\"true\"}")
    }

    @Test fun testLenientSerializerFalse() {
        val testHelper = JsonTestHelper()
        val serialized = testHelper.serialize(Lenient(false), Lenient::class.java)
        assertThat(serialized).isEqualTo("{\"foo\":\"false\"}")
    }

    @Test fun testLenientDeserializerTrue() {
        val testHelper = JsonTestHelper()
        val (foo) = testHelper.deserializeString(Lenient::class.java, "{\"foo\":\"true\"}")
        assertThat(foo).isEqualTo(true)
    }

    @Test fun testLenientDeserializerFalse() {
        val testHelper = JsonTestHelper()
        val (foo) = testHelper.deserializeString(Lenient::class.java, "{\"foo\":\"false\"}")
        assertThat(foo).isEqualTo(false)
    }
}
